Cubical/HITs/SetTruncation/Properties.agda

```diff
@@ -29,24 +29,40 @@ rec Bset f (squash₂ x y p q i j) =
   Bset _ _ (cong (rec Bset f) p) (cong (rec Bset f) q) i j
 
 rec2 : isSet C → (A → B → C) → ∥ A ∥₂ → ∥ B ∥₂ → C
-rec2 Cset f = rec (isSetΠ λ _ → Cset) λ x → rec Cset (f x)
+rec2 Cset f ∣ x ∣₂ ∣ y ∣₂ = f x y
+rec2 Cset f ∣ x ∣₂ (squash₂ y z p q i j) =
+  Cset _ _ (cong (rec2 Cset f ∣ x ∣₂) p) (cong (rec2 Cset f ∣ x ∣₂) q) i j
+rec2 Cset f (squash₂ x y p q i j) z =
+  Cset _ _ (cong (λ a → rec2 Cset f a z) p) (cong (λ a → rec2 Cset f a z) q) i j
+
+-- Old version:
+-- rec2 Cset f = rec (isSetΠ λ _ → Cset) λ x → rec Cset (f x)
 
 -- lemma 6.9.1 in HoTT book
 elim : {B : ∥ A ∥₂ → Type ℓ}
        (Bset : (x : ∥ A ∥₂) → isSet (B x))
-       (g : (a : A) → B (∣ a ∣₂))
+       (f : (a : A) → B (∣ a ∣₂))
        (x : ∥ A ∥₂) → B x
-elim Bset g ∣ a ∣₂ = g a
-elim Bset g (squash₂ x y p q i j) =
+elim Bset f ∣ a ∣₂ = f a
+elim Bset f (squash₂ x y p q i j) =
   isOfHLevel→isOfHLevelDep 2 Bset _ _
-    (cong (elim Bset g) p) (cong (elim Bset g) q) (squash₂ x y p q) i j
+    (cong (elim Bset f) p) (cong (elim Bset f) q) (squash₂ x y p q) i j
 
 elim2 : {C : ∥ A ∥₂ → ∥ B ∥₂ → Type ℓ}
         (Cset : ((x : ∥ A ∥₂) (y : ∥ B ∥₂) → isSet (C x y)))
-        (g : (a : A) (b : B) → C ∣ a ∣₂ ∣ b ∣₂)
+        (f : (a : A) (b : B) → C ∣ a ∣₂ ∣ b ∣₂)
         (x : ∥ A ∥₂) (y : ∥ B ∥₂) → C x y
-elim2 Cset f = elim (λ _ → isSetΠ (λ _ → Cset _ _))
-                    (λ a → elim (λ _ → Cset _ _) (f a))
+elim2 Cset f ∣ x ∣₂ ∣ y ∣₂ = f x y
+elim2 Cset f ∣ x ∣₂ (squash₂ y z p q i j) =
+  isOfHLevel→isOfHLevelDep 2 (λ a → Cset ∣ x ∣₂ a) _ _
+     (cong (elim2 Cset f ∣ x ∣₂) p) (cong (elim2 Cset f ∣ x ∣₂) q) (squash₂ y z p q) i j
+elim2 Cset f (squash₂ x y p q i j) z =
+  isOfHLevel→isOfHLevelDep 2 (λ a → Cset a z) _ _
+    (cong (λ a → elim2 Cset f a z) p) (cong (λ a → elim2 Cset f a z) q) (squash₂ x y p q) i j
```

Comment:
@ecavallo Didn't notice any massive speedup with this change... But I guess it can't hurt having a more direct definition